### PR TITLE
Add more minor antag objectives

### PR DESCRIPTION
**Pull request by AriaMoradi**
## About the PR Add more minor antagonist objectives

**Media**

- [x] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase
image

**Changelog**

:cl: ArMor
- add: Added more minor antagonist objectives

### CODE REVIEW

**Comment by DEATHB4DEFEAT** [comment]
good idea making an abstract type for the condition

**Review by DEATHB4DEFEAT — COMMENTED** [review]
looks pretty nice.

the objectives should probably be all in one localization file
since it seems you've renamed minor antag to freewill, you should also change the actual antag localizations to match

**Comment by DEATHB4DEFEAT** [comment]
YAML linter seems to be failing for reasons relating to this

**Review by DEATHB4DEFEAT — APPROVED** [review]

**Review by DEATHB4DEFEAT — APPROVED** [review]
amazing

File: Resources/Prototypes/SimpleStation14/Objectives/freewillObjectives.yml [hunk 1]
```diff
@@ -0,0 +1,106 @@
+- type: objective
+  id: FreewillAddictObjective
+  issuer: Your free will
+  difficultyOverride: 2.75
+  conditions:
+    - !type:GenericFreewillCondition
+      title: objective-condition-addict-title
+      description: objective-condition-addict-description
+      icon:
+        sprite: Clothing/Hands/Gloves/fingerless.rsi
```
Comment (DEATHB4DEFEAT) [inline Resources/Prototypes/SimpleStation14/Objectives/freewillObjectives.yml]: these all need icons

File: Resources/Prototypes/SimpleStation14/Objectives/freewillObjectives.yml [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+- type: objective
+  id: FreewillDealerObjective
+  issuer: Your free will
+  difficultyOverride: 2.75
+  conditions:
+    - !type:GenericFreewillCondition
+      title: objective-condition-dealer-title
+      description: objective-condition-dealer-description
+      icon:
+        sprite: Clothing/Hands/Gloves/fingerless.rsi
```
Comment (DEATHB4DEFEAT) [inline Resources/Prototypes/SimpleStation14/Objectives/freewillObjectives.yml]: icon, maybe a beaker

File: Resources/Prototypes/SimpleStation14/Objectives/freewillObjectives.yml [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+- type: objective
+  id: FreewillGraffitiArtistObjective
+  issuer: Your free will
+  difficultyOverride: 2.75
+  conditions:
+    - !type:GenericFreewillCondition
+      title: objective-condition-graffiti-artist-title
+      description: objective-condition-graffiti-artist-description
+      icon:
+        sprite: Clothing/Hands/Gloves/fingerless.rsi
```
Comment (DEATHB4DEFEAT) [inline Resources/Prototypes/SimpleStation14/Objectives/freewillObjectives.yml]: maybe add an icon, crayon box or something

File: Resources/Prototypes/SimpleStation14/Objectives/freewillObjectives.yml [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+- type: objective
+  id: FreewillLitterbugObjective
+  issuer: Your free will
+  difficultyOverride: 2.75
+  conditions:
+    - !type:GenericFreewillCondition
+      title: objective-condition-litterbug-title
+      description: objective-condition-litterbug-description
+      icon:
+        sprite: Clothing/Hands/Gloves/fingerless.rsi
```
Comment (DEATHB4DEFEAT) [inline Resources/Prototypes/SimpleStation14/Objectives/freewillObjectives.yml]: icon, trash?

File: Resources/Prototypes/SimpleStation14/Objectives/freewillObjectives.yml [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+- type: objective
+  id: FreewillObsessedObjective
+  issuer: Your free will
+  difficultyOverride: 2.75
+  conditions:
+    - !type:GenericFreewillCondition
+      title: objective-condition-obsessed-title
+      description: objective-condition-obsessed-description
+      icon:
+        sprite: Clothing/Hands/Gloves/fingerless.rsi
```
Comment (DEATHB4DEFEAT) [inline Resources/Prototypes/SimpleStation14/Objectives/freewillObjectives.yml]: icon, a heart?

File: Resources/Prototypes/SimpleStation14/Objectives/freewillObjectives.yml [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+- type: objective
+  id: FreewillPoorWorkerObjective
+  issuer: Your free will
+  difficultyOverride: 2.75
+  conditions:
+    - !type:GenericFreewillCondition
+      title: objective-condition-poor-worker-title
+      description: objective-condition-poor-worker-description
+      icon:
+        sprite: Clothing/Hands/Gloves/fingerless.rsi
```
Comment (DEATHB4DEFEAT) [inline Resources/Prototypes/SimpleStation14/Objectives/freewillObjectives.yml]: icon

File: Resources/Prototypes/SimpleStation14/Objectives/freewillObjectives.yml [hunk 7]
```diff
@@ -0,0 +1,106 @@
...
+- type: objective
+  id: FreewillSyndicateEnthusiastObjective
+  issuer: Your free will
+  difficultyOverride: 2.75
+  conditions:
+    - !type:GenericFreewillCondition
+      title: objective-condition-syndicate-enthusiast-title
+      description: objective-condition-syndicate-enthusiast-description
+      icon:
+        sprite: Clothing/Hands/Gloves/fingerless.rsi
```
Comment (DEATHB4DEFEAT) [inline Resources/Prototypes/SimpleStation14/Objectives/freewillObjectives.yml]: icon

File: Resources/Prototypes/SimpleStation14/Objectives/freewillObjectives.yml [hunk 8]
```diff
@@ -0,0 +1,106 @@
...
+
+- type: objective
+  id: FreewillTwitchyObjective
+  difficultyOverride: 2.75
+  conditions:
+    - !type:GenericFreewillCondition
+      title: objective-condition-twitchy-title
+      description: objective-condition-twitchy-description
+      icon:
+        sprite: Clothing/Hands/Gloves/fingerless.rsi
```
Comment (DEATHB4DEFEAT) [inline Resources/Prototypes/SimpleStation14/Objectives/freewillObjectives.yml]: icon

File: Resources/Prototypes/SimpleStation14/Objectives/objectiveGroups.yml [hunk 9]
```diff
@@ -1,13 +1,21 @@
...
-  id: MinorantagObjectiveGroup
+  id: MinorantagObjectiveGroup # Fixme: should be named `XxxGroups` according to convention
   weights:
-    MinorantagObjectiveGroupTheft: 1
+    FreewillObjectiveGroup: 1
 
 - type: weightedRandom
-  id: MinorantagObjectiveGroupTheft
+  id: FreewillObjectiveGroup
   weights:
```
Comment (DEATHB4DEFEAT) [inline Resources/Prototypes/SimpleStation14/Objectives/objectiveGroups.yml:10]: some of these should probably have slightly different weights

**Review by DEATHB4DEFEAT — CHANGES_REQUESTED** [review]
fix the conflicts

@Pspritechologist do what you need to also

File: Resources/Prototypes/SimpleStation14/Objectives/freewillObjectives.yml (hunk 2 above)
Comment (Pspritechologist) [reply Resources/Prototypes/SimpleStation14/Objectives/freewillObjectives.yml]: @DEATHB4DEFEAT Aw man, I didn't realize you suggested some, wish I'd looked at them :P 
For Dealer, I went with cash, standard green. Beaker would also be fun, but I wanted to emphasize that they don't _need_ to do drugs

File: Resources/Prototypes/SimpleStation14/Objectives/freewillObjectives.yml (hunk 3 above)
Comment (Pspritechologist) [reply Resources/Prototypes/SimpleStation14/Objectives/freewillObjectives.yml]: Rainbow spray can ;P

File: Resources/Prototypes/SimpleStation14/Objectives/freewillObjectives.yml (hunk 4 above)
Comment (Pspritechologist) [reply Resources/Prototypes/SimpleStation14/Objectives/freewillObjectives.yml]: Bag of chips, yep

File: Resources/Prototypes/SimpleStation14/Objectives/freewillObjectives.yml (hunk 5 above)
Comment (Pspritechologist) [reply Resources/Prototypes/SimpleStation14/Objectives/freewillObjectives.yml]: Ooo, that would've been good. I went with a blindfold...
Bit darker, I suppose :P

**Comment by Pspritechologist** [comment]
@AriaMoradi @DEATHB4DEFEAT 
Guess who got around to looking at the prrrrr
it me

Really good work actually, ArMor. Really nice way of handling the issue the Antag system poses :P
I've got all the sprites picked, requirements added, some code polished and everything all locally. I don't know the best way of getting that on Master though, since pushing it myself would leave this pr dead and I'd feel bad :P

**Comment by Pspritechologist** [comment]
@DEATHB4DEFEAT How're these changes looking? Good to merge this?

File: Resources/Changelog/SimpleStationChangelog.yml [hunk 10]
```diff
@@ -1265,3 +1265,11 @@ Entries:
         }
     id: 96
     time: "2023-02-25T01:45:13.0000000+00:00"
+  - author: "Pspritechologist and ArMor"
+    changes:
+      - {
+          message: "Finally properly implemented Minor Antags in the form of Flawed!",
```
Comment (DEATHB4DEFEAT) [inline Resources/Changelog/SimpleStationChangelog.yml]:
```suggestion
          message: "Finally properly implemented the 'Minor Antags', now with the name 'Flawed'!",
```

File: Resources/Prototypes/SimpleStation14/Objectives/objectiveGroups.yml [hunk 11]
```diff
@@ -1,19 +1,26 @@
-# Minor Antags
+# Flawed Antags
 - type: weightedRandom
-  id: MinorantagObjectiveGroup
+  id: FlawedObjectiveGroups
   weights:
-    MinorantagObjectiveGroupHarmless: 1
-    MinorantagObjectiveGroupDangerous: 0.3
+    FlawedObjectiveGroupMild: 1
+    FlawedObjevtiveGroupDanger: 0.6
```
Comment (DEATHB4DEFEAT) [inline Resources/Prototypes/SimpleStation14/Objectives/objectiveGroups.yml]:
```suggestion
    FlawedObjectiveGroupDanger: 0.6
```

File: Resources/Prototypes/SimpleStation14/Objectives/objectiveGroups.yml [hunk 12]
```diff
@@ -1,19 +1,26 @@
...
-    MinorPickpocketObjective: 1
+    FlawedAddictObjective: 0.6
+    FlawedGraffitiArtistObjective: 1
+    FlawedLitterbugObjective: 1
+    FlawedPickpocketObjective: 0.8
+    FlawedPoorWorkerObjective: 0.7
 
 - type: weightedRandom
-  id: MinorantagObjectiveGroupDangerous
+  id: FlawedObjevtiveGroupDanger
```
Comment (DEATHB4DEFEAT) [inline Resources/Prototypes/SimpleStation14/Objectives/objectiveGroups.yml]:
```suggestion
  id: FlawedObjectiveGroupDanger
```

**Review by DEATHB4DEFEAT — CHANGES_REQUESTED** [review]
"Flawed" is a weird name, would've preferred Freewill
Fix the file names
Move localizations to our subfolder

**Comment by DEATHB4DEFEAT** [comment]
reopen whenever you fix this all @Pspritechologist

**Pull request closed without merging**